src/treasury.cairo

```diff
@@ -0,0 +1,282 @@
+use starknet::ContractAddress;
+use governance::types::OptionType;
+
+#[starknet::interface]
+trait IGovernance<TContractState> {
+    fn claim(
+        ref self: TContractState, claimee: ContractAddress, amount: u128, proof: Array::<felt252>
+    );
+}
+
+#[starknet::interface]
+trait Itreasury<TContractState> {
+    fn send_tokens_to_address(
+        ref self: TContractState,
+        receiver: ContractAddress,
+        amount: u256,
+        tokenAddr: ContractAddress
+    ) -> bool;
+    fn update_governance_contract(ref self: TContractState, newGovernanceContract: ContractAddress);
+    fn update_AMM_contract(ref self: TContractState, newAMMContract: ContractAddress);
+    fn provide_liquidity_to_carm_AMM(
+        ref self: TContractState,
+        pooled_token_addr: ContractAddress,
+        quote_token_address: ContractAddress,
+        base_token_address: ContractAddress,
+        option_type: OptionType,
+        amount: u256
+    );
+    fn withdraw_liquidity(
+        ref self: TContractState,
+        pooled_token_addr: ContractAddress,
+        quote_token_address: ContractAddress,
+        base_token_address: ContractAddress,
+        option_type: OptionType,
+        lp_token_amount: u256
+    );
+    fn claim_airdrop_tokens(
+        ref self: TContractState, claimee: ContractAddress, amount: u128, proof: Array::<felt252>
+    );
+    fn get_governance_address(self: @TContractState) -> ContractAddress;
+    fn get_amm_address(self: @TContractState) -> ContractAddress;
+}
+
+#[starknet::contract]
+mod Treasury {
+    use core::starknet::event::EventEmitter;
+    use super::{OptionType, IGovernanceDispatcher, IGovernanceDispatcherTrait};
+    use core::num::traits::zero::Zero;
+    use openzeppelin::access::ownable::OwnableComponent;
+    use starknet::{ContractAddress, get_caller_address, get_contract_address};
+    use governance::traits::{
+        IERC20Dispatcher, IERC20DispatcherTrait, IAMMDispatcher, IAMMDispatcherTrait
+    };
+    component!(path: OwnableComponent, storage: ownable, event: OwnableEvent);
+
+    #[abi(embed_v0)]
+    impl OwnableImpl = OwnableComponent::OwnableImpl<ContractState>;
+
+    #[abi(embed_v0)]
+    impl OwnableCamelOnlyImpl =
+        OwnableComponent::OwnableCamelOnlyImpl<ContractState>;
+    impl InternalImpl = OwnableComponent::InternalImpl<ContractState>;
```

Comment:
Most of the calls like ```assert_only_owner()``` and ```initializer()``` are internal function calls, and without that declaration it would not be possible to call those functions.